### DISCUSSION TITLE
Unable to understand ECONNABORTED errorCode for XHR requests

### DISCUSSION THREAD

**Discussion by Suhasmv1998** [axios/axios#6257] [Q&A]
Hi,

We are getting ECONNABORTED error from Axios for our web App. When I look at axios documentation, it says that ECONNABORTED comes when "Request timed out due to exceeding timeout specified in axios configuration.". But the timeout we have set in Axios configuration is 30 seconds, but it is failing at 3 seconds (got to know from our telemetry) with this error code.  I suspect that user or browser itself has canceled the request and hence we were seeing this error. 

When I look at code, it says that ECONNABORTED is thrown in two cases, one when browser cancels the request and when there is timeout as well. Adding screenshots. Please let me know when exactly ECONNABORTED happens and is there a way I can repro it (so that I can debug)? Thank you in advance :)

![BrowserCancelation](https://github.com/axios/axios/assets/160602774/363fb163-d67c-43d8-b480-29b319fd1d12)
![Timeout error](https://github.com/axios/axios/assets/160602774/afcb5c8a-6b46-41b1-89b9-c8f173147482)

**Comment by Maxim-Mazurok** [comment]
I believe you can inspect the error message, it will be different in these two cases, for example, for the timeout:
![image](https://github.com/user-attachments/assets/0ef3a84d-14f1-4665-9f19-95c00f5f9526)

**Comment by DigitalBrainJS** [comment]
The documentation is a bit misleading. `ECONNABORTED` reflects the full spectrum of errors related to request cancellation (by default including timeouts), both inside the JS script and from the browser or its plugin. You need to set the `clarifyTimeoutError` flag in the `transitional` config to separate `ETIMEDOUT` timeout errors from other `ECONNABORTED`. Historically, Axios did not clarify them, which was fixed in version 1.x, but with the need to explicitly activate the new behavior. 

The `transitional` options are documented in the README.
```
  // transitional options for backward compatibility that may be removed in the newer versions
  transitional: {
    // silent JSON parsing mode
    // `true`  - ignore JSON parsing errors and set response.data to null if parsing failed (old behaviour)
    // `false` - throw SyntaxError if JSON parsing failed (Note: responseType must be set to 'json')
    silentJSONParsing: true, // default value for the current Axios version

    // try to parse the response string as JSON even if `responseType` is not 'json'
    forcedJSONParsing: true,

    // throw ETIMEDOUT error instead of generic ECONNABORTED on request timeouts
    clarifyTimeoutError: false,
  },
```
